R/create_analysis.R

```diff
@@ -0,0 +1,260 @@
+#' Create analysis from a design
+#'
+#' If not *data analysis plan (dap)* is provided, the analysis will run on the overall dataset, and all
+#' the grouping variables set.
+#'
+#' The *dap* should contains the following columns :
+#' - analysis_type: analysis type to be perform. At the moment mean, median, prop_select_one, and
+#' ratio are available.
+#' - analysis_var: analysis variable to be used as string.
+#' - group_var: The grouping variable as string. NA if there is no grouping variable. If a combination
+#' of grouping variable should be used together it should be 1 string character separated with a ",".
+#' i.e. c("admin1", "population") and "admin1, population" are different.
+#'    - c("admin1", "population") : will perform the analysis grouping once by admin1, and once by population
+#'    - "admin1, population" : will perform the analysis grouping once by admin1 and admin 2
+#' - level: confidence level to be used. If the column does not exists, .95 will be used.
+#' It can also include a column **level**, if not provided .95 will be set as default.
+#'
+#' If ratios have to be performed, the *dap* should include the following columns as well:
+#' - analysis_var_numerator analysis_var_denominator numerator_NA_to_0 filter_denominator_0
+#'
+#' @param .design Survey design object created with srvyr::as_survey or as_survey_design
+#' @param dap Data Analysis plan: Default is NULL. If provided it will be used to create the analysis.
```

Comment:
"One general comment, I tried to run the function with dap = NULL, and group_var = NULL. I received a warning message in the create_dap function, mainly the part where left_join(dap_dictionary)"
The warnings comes from the create_median, when having no numbers or low variations. 


"Also there are some NaN values introduced," :
Thoses values are only NAs (template was created with xlsxformfill), so it should make sense to have NaN

"Do we need these, or they can be filtered out?"
Did you mean in general "person_id, uuid, etc, types of columns"? I have added some filtering without the kobo in the create_loa
![image](https://github.com/impact-initiatives/analysistools/assets/79831444/1d146c98-dd11-4368-b46a-f4513404cb79)

For the NaN, I would rather keep the NaN in the results. It is easier to see a NaN and check the dataset rather than not having a results and having to find the results was removed because it was a NaN.

"I know that the naming is not the most urgent thing to do now, but I have faced many times confusion from the field regarding the naming of the Data DAP and the Research DAP. For this reason, I changed this DAP to DAF (Data Analysis Framework) and now there is no confusion at all."
Thanks for this, I have called it "list of analysis" as framework is for me building blocks rather than an iteration of the analysis to perform, but I hope that would remove any confusion with the one from the ToR.

Reply:
Well noted